File: app/controllers/bookings_controller.rb
class BookingsController < ApplicationController
  def index
    @bookings = Booking.paginate(page: params[:page], per_page: params[:per_page].presence || 4)
  end

  def new
    @booking = Booking.new
  end

  def create
    @booking = Booking.new(booking_params)

    if @booking.save
      redirect_to bookings_path
    else
      render :new
    end
  end

  private

  def booking_params
    params.require(:booking).permit(:day, :house_id,:number_of_guests)
  end
end

File: src/Controllers/BookingsController.php
<?php

namespace BookMyHouse\Controllers;

use BookMyHouse\Models\Booking;
use BookMyHouse\Models\House;

class BookingsController extends BaseController
{
    public function index(): void
    {
        $bookings = Booking::ordered();
        $this->render('bookings/index', ['bookings' => $bookings]);
    }

    public function new(): void
    {
        $booking = new Booking();
        $houses = House::ordered();
        $this->render('bookings/new', ['booking' => $booking, 'houses' => $houses]);
    }

    public function create(): void
    {
        $bookingData = [
            'day' => $this->getParam('day'),
            'house_id' => (int)$this->getParam('house_id'),
            'number_of_guests' => $this->getParam('number_of_guests') ? (int)$this->getParam('number_of_guests') : null
        ];

        $booking = new Booking($bookingData);
        $errors = $booking->validate();

        if (empty($errors) && $booking->save()) {
            $this->redirect('/');
        } else {
            $houses = House::ordered();
            $this->render('bookings/new', [
                'booking' => $booking, 
                'houses' => $houses, 
                'errors' => $errors
            ]);
        }
    }
}